### PR TITLE
ztest: Limit output when match pattern provided

### LINKED ISSUE #854: ztest.hpp matcher limit output
**Is your feature or enhancement request related to a problem or limitation? Please describe**

**Describe your enhancement idea**

If you run tests via `./ztest_runner.cpp` followed by a string that matches a test, only that test runs.  However, the full suite and test output is still printed, e.g. 

```
bash-4.3$ ./build-out/ztest_runner "should display a system symbol"          
======== TESTS ========
Running tests matching: should display a system symbol
uss
  copy uss tests
  chmod
    on a file
    on a directory
  chown
    on a file
    on a directory
 ...
```

it would be nice if this printed just the test output for the test that matched.

**Describe alternatives you've considered**

**Provide any additional context**

### PR DESCRIPTION
**Pull request by t1m0thyj**
**What It Does**

Resolves #854 - this builds on 8a131b5e4fb526bc36cc95880fc4ab0eb1b59123 to improve support for test patterns.

**How to Test**

Provide a pattern when running native tests, e.g.: `npm run z:test -- "'delete data sets'"`

Note that two sets of quotes are needed to prevent NPM from parsing each word in the pattern as a separate argument.

**Review Checklist**
I certify that I have:
- [ ] tested my changes
- [ ] added/updated automated tests
- [ ] updated the changelog
- [ ] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: native/c/test/ztest.hpp [hunk 1]
```diff
@@ -1130,7 +1147,20 @@ void describe(const std::string &description, Callable suite)
   {
     TEST_SUITE &current_suite = g.get_suites()[current_suite_idx];
     const std::vector<HOOK_WITH_OPTIONS> &after_all_hooks = current_suite.after_all_hooks;
-    if (!after_all_hooks.empty())
+
+    // Check if any non-skipped tests were added to this suite
+    bool has_real_tests = false;
+    for (const auto &test : current_suite.tests)
+    {
+      if (!test.skipped)
```
[the remark this answers is not in the document]
Comment (t1m0thyj) [reply native/c/test/ztest.hpp]: Resolved in https://github.com/zowe/zowex/pull/979/commits/13dc8ea53deca72cf6b13cb6fdba281b391c9609

**Comment by dkelosky** [comment]
Thank you for doing this - this is something we've needed for a long time!

I only tested this branch (not `main`).  This looks good but the summary is not correct:

```
bash-4.3$ ./build-out/ztest_runner "delete data sets"
======== TESTS ========
Running tests matching: delete data sets
zds
  delete data sets
    + PASS should fail if data set does not exist
    + PASS should fail if member does not exist
    + PASS should delete data set successfully when valid

======== TESTS SUMMARY ========
Suites:      1 skipped, 154 passed, 155 total
Tests:       32 skipped, 3 passed, 35 total
Time:        0.267s
```

Likewise:

```
bash-4.3$ ./build-out/ztest_runner "should fail if data set does not exist"
======== TESTS ========
Running tests matching: should fail if data set does not exist
zds
  delete data sets
    + PASS should fail if data set does not exist
  rename members
    + PASS should fail if data set does not exist

======== TESTS SUMMARY ========
Suites:      1 skipped, 154 passed, 155 total
Tests:       32 skipped, 2 passed, 34 total
Time:        0.071s
```

In both, `Suites` count is incorrect.

**Review by traeok — COMMENTED** [review]
One case that could cause issues related to `header_printed` flag - although its unlikely that someone would try to run skipped tests intentionally, consider this suite:

```cpp
describe("Mysuite", []() {
  afterAll([]() {
    // cleanup code - expects setup from beforeAll
  });
  
  xit("skipped test 1", []() { /* ... */ });
  xit("skipped test 2", []() { /* ... */ });
});
```

If you try to run tests matching `skipped test` they won't run (which is good, as they are marked skipped) - but the `afterAll` hook would still run because `header_printed` is set to `true`. Should we have a separate bool that signals whether any tests have run?

File: native/c/test/ztest.hpp [hunk 2]
```diff
@@ -1166,7 +1185,11 @@ void xit(const std::string &description, Callable)
     g.get_suites()[suite_idx].tests.push_back(tc);
   }
 
-  std::cout << get_indent(g.get_nesting()) << colors.skip << " SKIP " << description << std::endl;
+  if (g.matches_filter(description, g.get_matcher()))
```
Comment (traeok) [inline native/c/test/ztest.hpp:1213]: `matches_filter` was not introduced in this PR, but something to consider for the future: We should probably compile the pattern into a `std::regex` object once and store it as a member variable in the `Globals` singleton (e.g., inside `set_matcher()`). You can then catch invalid regexes upfront and reuse the compiled regex w/o having to recompile it for every test description.

**Comment by t1m0thyj** [comment]
@dkelosky Both cases should be fixed now 🙂 The number of suites we reported was always wrong compared to Jest - we were counting every `describe` block but are now including only top-level ones (one per test file).
image

image

File: native/c/test/ztest.hpp (hunk 2 above)
Comment (t1m0thyj) [reply native/c/test/ztest.hpp:1213]: Good idea 😋 Resolved in https://github.com/zowe/zowex/pull/979/commits/7f325f70f115f65b1f161f4e4cb4668f39690323 along with the edge case for `header_printed`

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
---

And I don't need the `'` single quotes now 🙏

**Review by dkelosky — APPROVED** [review]

**Review by traeok — APPROVED** [review]
Thanks for addressing my feedback, LGTM
